src/azure-cli-core/azure/cli/core/auth/identity.py

```diff
@@ -222,7 +227,7 @@ def get_user(self, user=None):
         return accounts
 
     def get_user_credential(self, username):
-        return UserCredential(self.client_id, username, **self._msal_app_kwargs)
+        return UserCredential(self.client_id, username, **self._msal_public_app_kwargs)
```

Comment:
We have no way of knowing whether device code flow is used in the initial `az login`. If so, and then we set `allow_broker` for `UserCredential`, token refreshing will fail (https://github.com/AzureAD/microsoft-authentication-library-for-python/issues/563).

Reply:
FYI: We have a solution now. [This MSAL PR](https://github.com/AzureAD/microsoft-authentication-library-for-python/pull/569) will fix the device code flow problem. Feel free to test it out